src/core/timesync.cpp

```diff
@@ -42,7 +42,7 @@ void Timesync::process_timesync(const mavlink_message_t& message)
                          _parent.get_autopilot_time().now().time_since_epoch())
                          .count();
 
-    if (timesync.tc1 == 0) {
+    if (timesync.tc1 == 0 && _timesync_locked) {
```

Comment:
May be
```suggestion
    if (timesync.tc1 == 0 && !_timesync_locked) {
```
?

Reply:
This looks right to me. We don't want to respond to timesyncs until our time is not updated with the autopilot time. Otherwise get_autopilottime will return system time at first.

Reply:
Okay, I understand now, thank you

Reply:
No, this is correct. We don't want the remote system to do a timesync until our own clock is stable.

Reply:
Ah, I think it is the naming of the variable that isn't clear. I will make it more obvious.